src/pixel-padding/index.tsx

```
@@ -0,0 +1,128 @@
+import { Devvit } from "@devvit/public-api";
+
+export const PixelPadding = (
+  props: Devvit.BlockComponentProps<PixelPaddingProps>,
+): JSX.Element => {
+  if (!Array.isArray(props.children) || !props.children.length) {
+    return <>{props.children}</>;
+  }
+
+  if (props.children.length > 1) {
+    return <>{...props.children}</>;
+  }
+
+  const childBlock = props.children[0];
+  if (typeof childBlock !== "object" || !childBlock) {
+    return <>{...props.children}</>;
```

:no_bell: maybe this conditional can be grouped with line 10. I think you're doing it for the type-check but it looks like we cast away the type on line 19 anyway.

---

Kinda on the fence about this one. Splitting the weird scenarios in groups makes it a bit easier for me to debug. I was also thinking about adding a custom console message to indicate that something is wrong and the component does not apply anything, but then decided not to

Curious if you'd like to see a log vs component silently ignoring it's duties

---

we've had issues with folks abusing the logger for debugging at the expense of every other contributor so I'd rather it be optional. maybe rename inspect to debug and flip it on then.